pyccel/codegen/printing/ccode.py

```diff
@@ -1120,42 +1118,76 @@ def _print_Assign(self, expr):
+        arg = rhs.arg
+        if rhs.rank > 1:
+            arg = functools.reduce(operator.concat, arg)
+        if isinstance(arg, Variable):
+            arg = self._print(arg)
+            if expr.lhs.is_stack_array:
+                cpy_data = self._init_stack_array(expr, rhs.arg)
+            else:
                 cpy_data = "memcpy({0}.{2}, {1}.{2}, {0}.buffer_size);".format(lhs, arg, dtype)
-                return '%s\n' % (cpy_data)
-            else :
-                arg = ', '.join(self._print(i) for i in arg)
-                dummy_array = "%s %s[] = {%s};\n" % (declare_dtype, dummy_array_name, arg)
-                cpy_data = "memcpy({0}.{2}, {1}, {0}.buffer_size);".format(lhs, dummy_array_name, dtype)
-                return  '%s%s\n' % (dummy_array, cpy_data)
-
-        if isinstance(rhs, (NumpyFull)):
-            code_init = ''
-            if rhs.fill_value is not None:
-                if isinstance(rhs.fill_value, Literal):
-                    dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-                    code_init = 'array_fill(({0}){1}, {2});'.format(dtype, self._print(rhs.fill_value), lhs)
-                else:
-                    code_init = 'array_fill({0}, {1});'.format(self._print(rhs.fill_value), lhs)
+            return '%s\n' % (cpy_data)
+        else :
+            arg = ', '.join(self._print(i) for i in arg)
+            dummy_array = "%s %s[] = {%s};\n" % (declare_dtype, dummy_array_name, arg)
+            if expr.lhs.is_stack_array:
+                cpy_data = self._init_stack_array(expr, dummy_array_name)
             else:
-                return ''
-            return '{}\n'.format(code_init)
+                cpy_data = "memcpy({0}.{2}, {1}, {0}.buffer_size);".format(self._print(lhs), dummy_array_name, dtype)
+            return  '%s%s\n' % (dummy_array, cpy_data)
 
-        rhs = self._print(rhs)
-        return '{} = {};'.format(lhs, rhs)
+    def print_NumpyFull(self, expr):
+        rhs = expr.rhs
+        lhs = expr.lhs
+        code_init = ''
+        if lhs.is_stack_array:
+            declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+            lenght = '*'.join(self._print(i) for i in lhs.shape)
+            buffer_array = "({declare_dtype}[{lenght}]){{}}".format(declare_dtype = declare_dtype, lenght=lenght)
+            code_init += self._init_stack_array(expr, buffer_array)
+        if rhs.fill_value is not None:
+            if isinstance(rhs.fill_value, Literal):
+                dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+                code_init += 'array_fill(({0}){1}, {2});\n'.format(dtype, self._print(rhs.fill_value), self._print(lhs))
+            else:
+                code_init += 'array_fill({0}, {1});\n'.format(self._print(rhs.fill_value), self._print(lhs))
+        return '{}'.format(code_init)
+
+    def _init_stack_array(self, expr, buffer_array, empty=False):
```

Comment:
Please add a docstring to this function. Codacy doesn't complain in this file as all the `_print_X` functions would have nearly the same docstring, but any other functions should be properly documented